src/query/query_executor.c

```diff
@@ -6718,14 +6719,15 @@ qexec_open_scan (THREAD_ENTRY * thread_p, ACCESS_SPEC_TYPE * curr_spec, VAL_LIST
 	}
       break;
     case TARGET_DBLINK:
+      host_vars.count = curr_spec->s.dblink_node.host_var_count;
+      host_vars.index = curr_spec->s.dblink_node.host_var_index;
       error_code =
 	scan_open_dblink_scan (thread_p, s_id,
 			       curr_spec->single_fetch, scan_op_type,
 			       curr_spec->s.dblink_node.conn_url,
 			       curr_spec->s.dblink_node.conn_user,
 			       curr_spec->s.dblink_node.conn_password,
-			       curr_spec->s.dblink_node.conn_sql,
-			       vd, val_list, curr_spec->s.dblink_node.dblink_regu_list_pred, curr_spec->where_pred);
+			       curr_spec->s.dblink_node.conn_sql, vd, val_list, &host_vars, curr_spec->where_pred);
```

Comment:
how about just passing `curr_spec->s.dblink_node` as an argument of `scan_open_dblink_scan()`?

Reply:
Ok I'll try.